src/runtime/database.cpp

```diff
@@ -365,6 +368,19 @@ std::string Database::open(bool wait, bool memory, bool tty) {
     "select distinct user.job_id as user, used.job_id as used"
     "  from filetree user, filetree used"
     "   where user.access=1 and user.file_id=used.file_id and used.access=2";
+  const char *sql_get_job_visualization =
+    "select j.job_id, j.label, j.directory, j.commandline, j.environment, j.stack, j.stdin, j.starttime, j.endtime, j.stale, r.time, r.cmdline, s.status, s.runtime, s.cputime, s.membytes, s.ibytes, s.obytes"
+    " from  jobs j left join stats s on j.stat_id=s.stat_id join runs r on j.run_id=r.run_id"
+    " where instr(j.commandline, '<source>') <= 0"
+    " and instr(j.commandline, '<mkdir>') <= 0"
+    " and instr(j.commandline, '<write>') <= 0"
+    " and instr(j.commandline, '<hash>') <= 0";
```

Comment:
Maybe just compare the prefix of j.commandline?

Reply:
Done